### ISSUE TITLE
False Positive for UnnecessaryLambda when using Predicate

### ISSUE THREAD

**Issue by codylerum** [google/error-prone#1372]
### Description of the problem / feature request:

> False Warning moving from 2.3.3 to 2.34 SNAPSHOT for UnnecessaryLambda when using a predicate in a Stream filter.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

> Warning:(475, 52) java: [UnnecessaryLambda] Returning a lambda from a helper method or saving it in a constant is unnecessary; prefer to implement the functional interface method directly and use a method reference instead.
    (see https://errorprone.info/bugpattern/UnnecessaryLambda)
  Did you mean 'final BigDecimal ccsQuantity = s.getEntries().stream().filter(QuoteTools::trunkProduct).map(QuoteSiteDetailEntry::getQuantity).reduce(BigDecimal.ZERO, BigDecimal::add);'?

Stream
```
final BigDecimal ccsQuantity = s.getEntries().stream().filter(trunkProduct()).map(QuoteSiteDetailEntry::getQuantity).reduce(BigDecimal.ZERO, BigDecimal::add);
```
Predicate
```
 private static Predicate<QuoteSiteDetailEntry> trunkProduct() {
        return p -> Objects.nonNull(p.getProduct().getType()) && p.getProduct().getType().isTrunkProduct();
}
```

I do not believe it is possible to convert trunkProduct to a method refernce like `QuoteTools::trunkProduct`

### What version of Error Prone are you using?

> 2.3.4-SNAPSHOT

###  Have you found anything relevant by searching the web?

> No

**Comment by tbroyer** [comment]
How about the following?

```java
private static boolean trunkProduct(QuoteSiteDetailEntry p) {
  return Objects.nonNull(p.getProduct().getType()) && p.getProduct().getType().isTrunkProduct();
```

**Comment by lowasser** [comment]
The intent of this check is definitely that you do the rewrite @tbroyer suggested, after which you can use the method reference as described.  I'm not sure the suggested fix makes that clear, though.
